response_operations_ui/views/case.py

```diff
@@ -79,3 +94,22 @@ def get_timestamp_for_completed_case_event(case_id):
     timestamp = case_events[last_index]['createdDateTime'].replace("T", " ").split('.')[0]
 
     return get_formatted_date(timestamp)
+
+
+def get_case_event_for_seft_or_eq(case_events):
+    case_event = [event for event in case_events if event['category'] in SUCCESSFUL_CASE_EVENTS]
+    return case_event
+
+
+def get_user_from_case_events(case_events):
+    first_case_event = len(case_events) - 1
+    if case_events[first_case_event]['metadata']:
```

Comment:
shouldn't you check that there's a `partyId` in the `metadata` at this point?

Reply:
The reason I did it like this was because if I look for `partyId` at the same time as `metadata`, it would raise two types of error. I thought this way would be more readable for anyone looking at the code. It would skip it if there was no `metadata` and if no `partyId` it would raise a `KeyError`

Reply:
sounds good